### PR TITLE
(A little more than) Search bar setup

### PR DESCRIPTION
**Pull request by NSMcEnroe**
**Features:** -Regular search bar functionality, with results page
(HTML and CSS are far from perfect at the moment, but Will suggests we clean that all up with Bootstrap once we have functionality down)

-Advanced search bar paths, and the "normal search" can be done at the moment
(The advanced search part is not complete, but nothing should break atm, it will just do a regular search, albeit a different path)

**Things to Note:** 

**Tests:** I believe I have a good chunk of the basic tests down, but there may be some sad paths, edge cases, and basic functionality I have not tested yet.  Feel free to notify me or add some if this is the case.

I also do have a list of possible bugs and user-friendly features I would like to include in my backlog, so let me know if you want to see those.

**UI:**  Only using the card feature at the moment.  We will probably eventually just use a basic template for our website.

**Header:** I would eventually like a drop down search bar, but this will suffice for now.

**User interaction:** There is currently no interactions between the user and the search bar, so this will need to be connected once we add save functionality onto the app

### CODE REVIEW

**Comment by WillWeston94** [comment]
This is looking like some solid work so far! Here are a few thoughts:

1. **Commit Descriptions**: The commit descriptions are clear and give a good indication of what was changed. 

2. **Features**: I see that you've added both regular and advanced search functionalities. It's good to see that you're thinking ahead by adding paths for the advanced search even if the actual functionality isn't complete. This will make future integration smoother.

3. **Tests**: It's great that you've already begun testing, especially as these features are still in development. Once CircleCI is up today we will be able to integrate the PRs with automatic testing capability as well for all previous tests to make sure they still work! 

4. **Header**: The dropdown search bar would be a nice touch for improving the user experience. 

-Im going to add the dropdown bar suggestion to the project board so we dont lost track of the extension idea! 

Approved ✅

File: app/facades/recipe_facade.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(search)
+    @search = search
+  end
+
+  def recipes
+    data = RecipeDatabaseService.new.recipes_by_keyword(@search)
+
+    data[:data].map do |recipe_data|
+      Recipe.new(recipe_data)
+    end
```
Comment (kicotiel) [inline app/facades/recipe_facade.rb:12]: One consideration here, and it may not be something we run into. If you need to call recipes multiple times within a view, each of those times would then hit the API and map out the data that was already mapped in the previous call of recipes.

_IF_ that becomes the case, it may be worth considering registering `@recipes` as an instance variable on the RecipeFacade, and just set it equal to your existing method. That way when it initializes, it calls the method once, and then that result is just stored in the instance variable rather than calling it each time separately.

Just a small efficiency thing to consider if we ever have to call it multiple times :) Great as is though.

**Review by kicotiel — APPROVED** [review]
Yeah this looks great. See the one comment I made on an instance variable in the Facade, but that's not important just yet and may never become a thing.

The only other item I'd mention, is whoever pushes next should .gitignore the coverage folder, delete it, and then push. That'll get rid of our coverage stuff being in the repo, which is totally not needed and just bulks it up lol.
